apis/core/v1alpha1/function_types.go

```diff
@@ -134,23 +131,39 @@ type FunctionSpec struct {
 	Version *string `json:"version,omitempty"`
 	// Function image name
 	Image string `json:"image"`
+	// ImageCredentials references a Secret that contains credentials to access
+	// the image repository.
+	//
+	// +optional
+	ImageCredentials *v1.LocalObjectReference `json:"imageCredentials,omitempty"`
 	// The port on which the function will be invoked
 	Port *int32 `json:"port,omitempty"`
-	// Information needed to build a function
+	// Information needed to build a function. If the `Build` is nil, it will skip the build step.
 	Build *BuildImpl `json:"build,omitempty"`
-	// Information needed to run a function
+	// Information needed to run a function. If the `Serving` is nil, it will skip the serving step.
```

Comment:
// Information needed to run a function. The serving step will be skipped if `Serving` is nil.